pilot/pkg/controllers/ipallocate/ipallocate.go

```diff
@@ -344,14 +345,21 @@ func (c *IPAllocator) statusPatchForAddresses(se *networkingv1.ServiceEntry, for
 	}
 
 	existingHostAddresses := autoallocate.GetHostAddressesFromServiceEntry(se)
-	existingAddresses := []netip.Addr{}
 	hostsWithAddresses := sets.New[string]()
+	hostsInSpec := sets.New[string]()
+
+	for _, host := range maps.Keys(existingHostAddresses) {
+		hostsWithAddresses.Insert(host)
```

Comment:
Why split this from L360, and maps.Keys is inefficient compared to directly iterate on the map

Reply:
i was trying to return much earlier. Now i realized it is not possible. I will revert this change